src/Sportradar.OddsFeed.SDK/Entities/REST/Internal/EntitiesImpl/Competitor.cs

```diff
@@ -508,6 +514,23 @@ private CompetitorCI GetOrLoadCompetitor()
             return _competitorCI;
         }
 
+        private async Task<CompetitorCI> GetOrLoadCompetitorAsync()
```

Comment:
This is a combination of `GetOrLoadCompetitor` and `LoadCompetitorProfileInCache` - The approach was to make one method which results in less changes